File: Binary Search.java

public class Solution { 

    public static int binarySearch(int[] arr, int x) {
        int start = 0;
        int end = arr.length - 1;
        
        while (start <= end){
            int mid = (start + end)/2;
            if (x == arr[mid]){
                return mid;
            } else if (x < mid){
                end = mid - 1;
            } if (x > mid){
                end = mid + 1;
            }
          
            
        }
        return -1;
        
    }

}

File: binarySearch.c
#include <stdio.h>

int binarySearch(int arr[], int value, int left, int right)
{
    if (left > right)
    {
        return -1;
    }
    int mid = (left + right) / 2;

    if (arr[mid] == value)
    {
        return mid;
    }
    else if (arr[mid] < value)
    {
        return binarySearch(arr, value, mid + 1, right);
    }
    else
    {
        return binarySearch(arr, value, left, mid - 1);
    }
}

int main()
{
    int array[] = {1, 4, 6, 8, 9, 10, 15, 36, 45, 97};
    int n = 10;

    int idx = binarySearch(array, 36, 0, n - 1);

    printf("%d", idx);

    return 0;
}
